app/src/main/kotlin/com/simplemobiletools/clock/services/TimerService.kt

```diff
@@ -34,18 +37,43 @@ class TimerService : Service() {
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
         super.onStartCommand(intent, flags, startId)
-
-        val formattedDuration = config.timerSeconds.getFormattedDuration()
-        startForeground(TIMER_RUNNING_NOTIF_ID, notification(formattedDuration))
-
+        updateNotification()
+        startForeground(TIMER_RUNNING_NOTIF_ID, notification(getString(R.string.app_name), getString(R.string.timer_notification_msg), INVALID_TIMER_ID))
         return START_NOT_STICKY
     }
 
+    private fun updateNotification() {
+        timerHelper.getTimers { timers ->
+            val runningTimers = timers.filter { it.state is TimerState.Running }
+            if (runningTimers.isNotEmpty()) {
+                val firstTimer = runningTimers.first()
+                val formattedDuration = (firstTimer.state as TimerState.Running).tick.getFormattedDuration()
+                val contextText = when {
+                    runningTimers.size > 1 -> {
+                        getString(R.string.timer_multiple_notification_msg, runningTimers.size)
+                    }
+                    firstTimer.label.isNotEmpty() -> {
+                        getString(R.string.timer_single_notification_label_msg, firstTimer.label)
+                    }
+                    else -> {
+                        getString(R.string.timer_single_notification_msg, runningTimers.size)
+                    }
```

Comment:
you can move such 1 liners into 1 line, like "else -> getString(R.string.timer_single_notification_msg, runningTimers.size)"